jerry-core/vm/vm.c

```diff
@@ -65,30 +66,41 @@ static ecma_compiled_code_t *__program = NULL;
  */
 static ecma_value_t
 vm_op_get_value (ecma_value_t object, /**< base object */
-                 ecma_value_t property, /**< property name */
-                 bool is_strict) /**< strict mode */
+                 ecma_value_t property) /**< property name */
 {
   if (unlikely (ecma_is_value_undefined (object) || ecma_is_value_null (object)))
   {
     return ecma_raise_type_error (ECMA_ERR_MSG (""));
   }
 
-  ecma_value_t completion_value = ecma_make_simple_value (ECMA_SIMPLE_VALUE_EMPTY);
+  ecma_value_t completion_value = ecma_op_to_string (property);
 
-  ECMA_TRY_CATCH (property_val,
-                  ecma_op_to_string (property),
-                  completion_value);
+  if (ecma_is_value_error (completion_value))
+  {
+    return completion_value;
+  }
 
-  ecma_string_t *property_p = ecma_get_string_from_value (property_val);
+  ecma_string_t *property_name_p = ecma_get_string_from_value (completion_value);
 
-  ecma_reference_t reference = ecma_make_reference (object, property_p, is_strict);
+  if (ecma_is_value_object (object))
+  {
+    ecma_object_t *object_p = ecma_get_object_from_value (object);
 
-  completion_value = ecma_op_get_value_object_base (reference);
+    ecma_property_t *property_p = ecma_lcache_lookup (object_p, property_name_p);
 
-  ecma_free_reference (reference);
+    if (property_p != NULL &&
+        ECMA_PROPERTY_GET_TYPE (property_p) == ECMA_PROPERTY_TYPE_NAMEDDATA)
+    {
+      completion_value = ecma_fast_copy_value (ecma_get_named_data_property_value (property_p));
 
-  ECMA_FINALIZE (property_val);
+      ecma_deref_ecma_string (property_name_p);
+      return completion_value;
+    }
+  }
+
+  completion_value = ecma_op_get_value_object_base (object, property_name_p);
```

Comment:
I see now. You reused the same variable, this is why it's name is so general. Not sure it's worth it, but it won't be an objection against the merge from my side. :)